compliance/collection/main.go

```diff
@@ -127,6 +130,30 @@ func manageStream(ctx context.Context, cli sensor.ComplianceServiceClient, sig *
 	}
 }
 
+func manageNodeScanLoop(ctx context.Context, cli sensor.ComplianceServiceClient, scanner nodescanv2.NodeScanner, rescanInterval time.Duration) {
+	t := time.NewTicker(5 * time.Second)
+	log.Infof("Node Rescan interval: %v", rescanInterval)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-t.C:
+			client, _, err := initializeStream(ctx, cli)
```

Comment:
I wonder if we should throw this inside of `manageStream` so we can just reuse the same stream and use the same stop signals

Reply:
We tried this and this is not trivial, because the `runRecv` blocks. I have a variation on this here https://github.com/stackrox/stackrox/pull/3548 but it also uses two loops - one for sending and one for receiving.

Reply:
In [a35000a](https://github.com/stackrox/stackrox/pull/3399/commits/a35000a42f261cf1a0e135f0714354e3ccf49198), we applied the idea from #3548 and now, we have separate loops for sending and receiving